tests/omd.ml

```diff
@@ -7,8 +7,9 @@ let normalize_html s =
 
 let with_open_in fn f =
   let ic = open_in fn in
-  Fun.protect ~finally:(fun () -> close_in_noerr ic)
-    (fun () -> f ic)
+  let result = f ic in
+  close_in_noerr ic;
+  result
```

Comment:
As this is used in the main library code, let's be a bit more careful, and use the following definition for `protect`:
```
let protect ~finally f =
  match f () with
  | exception e -> finally (); raise e
  | r -> finally (); r
```

Reply:
Yes I think it should be protected in both cases. Btw I tried to create a compact module to share between `extract_tests` and `omd` and got this error.
```
Error: Module "Compat" is used in several stanzas:
- tests/dune:1
- tests/dune:13
To fix this error, you must specify an explicit "modules" field in every
library, executable, and executables stanzas in this dune file. Note that
each module cannot appear in more than one "modules" field - it must belong
to a single library or executable.
```

I think it's not too bad to have this short function in both files as an alternative.

Reply:
Yes, it OK to duplicate the definition.